### PR TITLE
boot_nic_with_intel_iommu: Add test case

### PR DESCRIPTION
**Pull request by xiywang**
Boot guest with intel_iommu=on in guest kernel cmd line

id: 1691263

Signed-off-by: Xiyue Wang <xiywang@redhat.com>

### CODE REVIEW

**Comment by ybduan** [comment]
Check failed for "Unused import logging".

**Comment by xiywang** [comment]
Depends on: https://github.com/avocado-framework/avocado-vt/pull/2010

**Comment by xiywang** [comment]
(1/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.boot_nic_with_iommu_platform.q35: PASS (69.99 s)
(2/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.boot_nic_with_iommu_platform.q35: PASS (165.42 s)

File: qemu/tests/cfg/boot_nic_with_intel_iommu.cfg [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    machine_type_extra_params = "kernel-irqchip=split"
+    virtio_dev_disable_legacy = on
+    virtio_dev_disable_modern = off
+    virtio_dev_iommu_platform = on
+    virtio_dev_ats = on
+    vhostforce = on
+    pci_controllers += "pcie_nic"
+    type_pcie_nic = "ioh3420"
+    pci_bus_nic1 = pcie_nic
+    Linux:
```
Comment (lolyu) [inline qemu/tests/cfg/boot_nic_with_intel_iommu.cfg]: is this `Linux:` necessary?

Comment (xiywang) [reply qemu/tests/cfg/boot_nic_with_intel_iommu.cfg]: I think so. No need to set this value for Windows. Besides, the param is added to the linux guest kernel cmd line. 
Anyway, if we remove "Linux:", this case also works fine.

File: qemu/tests/cfg/boot_nic_with_intel_iommu.cfg [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    type = boot_nic_with_intel_iommu
+    extra_params = "-device intel-iommu,intremap=on,device-iotlb=on"
+    machine_type_extra_params = "kernel-irqchip=split"
+    virtio_dev_disable_legacy = on
+    virtio_dev_disable_modern = off
+    virtio_dev_iommu_platform = on
+    virtio_dev_ats = on
+    vhostforce = on
+    pci_controllers += "pcie_nic"
+    type_pcie_nic = "ioh3420"
```
Comment (vivianQizhu) [inline qemu/tests/cfg/boot_nic_with_intel_iommu.cfg]: Why use `ioh3420`? I think we abandon it and use `pcie-root-port` instead now.

Comment (xiywang) [reply qemu/tests/cfg/boot_nic_with_intel_iommu.cfg]: The developer told me the cmd line to test this case, I just followed his suggestion.

Comment (vivianQizhu) [reply qemu/tests/cfg/boot_nic_with_intel_iommu.cfg]: @xiywang Well, I believe the information is out of date, `ioh3420` is an unofficial one before `pcie-root-port` being provided.

Comment (pezhang) [reply qemu/tests/cfg/boot_nic_with_intel_iommu.cfg]: I agree, we should use pcie-root-port here.

**Comment by xiywang** [comment]
@vivianQizhu @pezhang Updated.

File: qemu/tests/cfg/boot_nic_with_intel_iommu.cfg [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    extra_params = "-device intel-iommu,intremap=on,device-iotlb=on"
+    machine_type_extra_params = "kernel-irqchip=split"
+    virtio_dev_disable_legacy = on
+    virtio_dev_disable_modern = off
+    virtio_dev_iommu_platform = on
+    virtio_dev_ats = on
+    vhostforce = on
+    pci_controllers += "pcie_nic"
+    type_pcie_nic = "pcie-root-port"
+    pci_bus_nic1 = pcie_nic
```
Comment (vivianQizhu) [inline qemu/tests/cfg/boot_nic_with_intel_iommu.cfg]: You don't need these three line, q35 has default topology to plug pcie device under pcie-root-port, you will get correct command line without them.

Comment (xiywang) [reply qemu/tests/cfg/boot_nic_with_intel_iommu.cfg]: Updated.

**Comment by pezhang** [comment]
Test pass.

 (1/1) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.boot_nic_with_intel_iommu.q35: PASS (83.73 s)


Looks good to me.
Acked by: Pei Zhang pezhang@redhat.com

**Pull request merged**